doc/man7/flux-jobtap-plugins.rst

```diff
@@ -251,6 +251,31 @@ or via configuration (See :ref:`configuration` below)
    ]
 
 
+.. _perilogs:
+
+PROLOG AND EPILOG ACTIONS
+=========================
+
+Plugins that need to perform asynchronous tasks for jobs after an ``alloc``
+event but before the job is running, or after a ``finish`` event but before
+resources are freed to the scheduler can make use of job manager prolog or
+epilog actions.
+
+To initiate a prolog action, a plugin should call the function 
+``flux_jobtap_prolog_start()``. This will block the job from starting even
+after resources have been assigned until a corresponding call to
+``flux_jobtap_prolog_finish()`` has been called. While the status of
```

Comment:
That's my understand - see `perilog-test.c` in  7901ffd

Reply:
Sorry, I was out on a ride.

Yes, any plugin that needs to do some work before the job is handed off to the execution system should call `flux_jobtap_prolog_start()` in either the `job.state.run` or `job.event.alloc` callback (note however, that normally jobtap plugins do not get the `job.event.*` callbacks without first "subscribing" to job events with `flux_jobtap_job_subscribe()`) Once the work is finished, the plugin should call `flux_jobtap_prolog_finish()`. Once all outstanding prolog actions have finished, only then will the start request be sent along to the execution system.

Currently the outstanding prolog/epilog actions are simply counted, and nothing is done with the `status` in the `*_finish()` functions (the action itself should determine if a job exception should be raised), and therefore those context fields are purely informative.


Soon I'll have a PR with an example that optionally runs a prolog and/or epilog script using this facility, so that may be a good example.

Reply:
Might be good to expand the doc a little to incorporate which callbacks would call `flux_jobtap_prolog_start()` and `flux_jobtap_epilog_start()`, and the caveat about subscription for those events.

Reply:
Good idea! I'll do that.